docs/src/examples/IEGT_example1.jl

```diff
@@ -0,0 +1,91 @@
+# # Foundations of Bilevel Programming: Example 9
+# This example is from the book Investment in Electricity Generation and Transmission. [url](https://www.springer.com/gp/book/9783319294995)
```

Comment:
Since this example is more realistic, it needs a description of the physical problem. What the variables represent, etc.

Some (not all. I can do better) of the JuMP tutorials do this:
https://jump.dev/JuMP.jl/stable/tutorials/Nonlinear%20programs/space_shuttle_reentry_trajectory/
https://jump.dev/JuMP.jl/stable/tutorials/Mixed-integer%20linear%20programs/sudoku/

Reply:
Yes, we need a bit of story telling, nothing complex. something like:

In this example we consider a power system with two existing generators and one being planned.
Their costs are... and existing generators have... available capacity.
The problem is to optimize the size of the new generators given that....